internal/auth0/organization/resource_connection_test.go

```diff
@@ -117,30 +183,80 @@ func TestAccOrganizationConnection(t *testing.T) {
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr("data.auth0_organization.my_org", "connections.#", "2"),
-					resource.TestCheckResourceAttr("auth0_organization_connection.my_org_conn_1", "assign_membership_on_login", "true"),
-					resource.TestCheckTypeSetElemAttrPair("auth0_organization_connection.my_org_conn_1", "organization_id", "auth0_organization.my_org", "id"),
-					resource.TestCheckTypeSetElemAttrPair("auth0_organization_connection.my_org_conn_1", "connection_id", "auth0_connection.my_connection_1", "id"),
-					resource.TestCheckResourceAttr("auth0_organization_connection.my_org_conn_2", "assign_membership_on_login", "true"),
-					resource.TestCheckTypeSetElemAttrPair("auth0_organization_connection.my_org_conn_2", "organization_id", "auth0_organization.my_org", "id"),
-					resource.TestCheckTypeSetElemAttrPair("auth0_organization_connection.my_org_conn_2", "connection_id", "auth0_connection.my_connection_2", "id"),
+					resource.TestCheckResourceAttr("auth0_organization_connection.my_org_db_conn", "assign_membership_on_login", "true"),
+					resource.TestCheckTypeSetElemAttrPair("auth0_organization_connection.my_org_db_conn", "organization_id", "auth0_organization.my_org", "id"),
+					resource.TestCheckTypeSetElemAttrPair("auth0_organization_connection.my_org_db_conn", "connection_id", "auth0_connection.my_db_connection", "id"),
+					resource.TestCheckResourceAttr("auth0_organization_connection.my_org_enterprise_conn", "assign_membership_on_login", "true"),
+					resource.TestCheckTypeSetElemAttrPair("auth0_organization_connection.my_org_enterprise_conn", "organization_id", "auth0_organization.my_org", "id"),
+					resource.TestCheckTypeSetElemAttrPair("auth0_organization_connection.my_org_enterprise_conn", "connection_id", "auth0_connection.my_enterprise_connection", "id"),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(testAccOrganizationConnectionsDelete, testName),
+			},
+			{
+				RefreshState: true,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.auth0_organization.org_data", "connections.#", "0"),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(testAccOrganizationConnectionSetShowAsButton, testName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.auth0_organization.org_data", "connections.#", "1"),
+					resource.TestCheckResourceAttr("data.auth0_organization.org_data", "connections.0.show_as_button", "true"),
+					resource.TestCheckResourceAttr("auth0_organization_connection.my_org_db_conn", "show_as_button", "true"),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(testAccOrganizationConnectionClearShowAsButton, testName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.auth0_organization.org_data", "connections.#", "1"),
+					resource.TestCheckResourceAttr("data.auth0_organization.org_data", "connections.0.show_as_button", "false"),
```

Comment:
Let's have assertions for all the parameters, even if we don't set them in the test config, as the state will do so for us.